test/sqllogictest/arrays.slt

```diff
@@ -170,6 +170,72 @@ SELECT unnest::text FROM unnest(NULL::int[])
 query error Cannot call function unnest\(unknown\): arguments cannot be implicitly cast to any implementation's parameters; try providing explicit casts
 SELECT * FROM unnest(NULL)
 
+# array_agg
+
+query T
+SELECT array_agg(a) FROM (SELECT 1 AS a WHERE false)
+----
+NULL
+
+query T
+SELECT array_agg(1)
+----
+{1}
+
+statement ok
+CREATE TABLE t1 (a int)
+
+statement ok
+INSERT INTO t1 VALUES (1), (2), (3), (NULL), (NULL)
+
+query T
+SELECT array_agg(a) FROM (select a from t1 where a IS NOT NULL)
+----
+{1,2,3}
+
+query T
+SELECT array_agg(a) FROM (select a from t1 where a IS NULL)
+----
+{NULL,NULL}
+
+query T
+SELECT array_agg(a) FROM t1
+----
+{NULL,NULL,1,2,3}
+
+query T
+SELECT array_agg(a::text) FROM t1
+----
+{NULL,NULL,1,2,3}
+
+query T
+SELECT array_agg(a) FILTER (WHERE a IS NOT NULL) FROM t1
+----
+{1,2,3}
+
+query T
+SELECT array_agg(a) FILTER (WHERE a IS NULL) FROM t1
+----
+{NULL,NULL}
+
+query error arguments cannot be implicitly cast to any implementation's parameters
+SELECT array_agg(1, 2)
+
+statement ok
+CREATE TABLE t2 (a int, b date)
+
+statement ok
+INSERT INTO t2 VALUES (1, date '2020-01-01'), (NULL, date '2020-01-02')
+
+query error arguments cannot be implicitly cast to any implementation's parameters
+SELECT array_agg((a, b)) FROM t2
+
+query error arguments cannot be implicitly cast to any implementation's parameters
+SELECT array_agg((a, b)), array_agg(a), array_agg(b) FROM t2
+
+query error not yet supported
+SELECT array_agg(ARRAY[1])
+
```

Comment:
I want to believe there's a somewhat simpler way to write this, but can we get either a passing test or an expected failure on something that uses the result of `unnest` as the input to `array_agg`? e.g.

```
select array_agg(unnest) FROM (SELECT NULL) x JOIN LATERAL unnest(ARRAY[1,2,NULL]) ON true;
```

Apparently the lateral join's required by PG.

Reply:
Cool we give the same output (except null ordered first) as pg. Test added.